HLTriggerOffline/Egamma/src/EmDQMPostProcessor.cc

```diff
@@ -359,6 +368,39 @@ TProfile* EmDQMPostProcessor::dividehistos(DQMStore::IBooker & ibooker, DQMStore
 
 //----------------------------------------------------------------------
 
+TH2F* EmDQMPostProcessor::dividehistos2D(DQMStore::IBooker & ibooker, DQMStore::IGetter & igetter, const std::string& numName, const std::string& denomName, const std::string& outName,const std::string& label,const std::string& titel){
+  //std::cout << numName <<std::endl;
+
+  TH2F* num = get2DHistogram(ibooker,igetter,numName);
+
+  //std::cout << denomName << std::endl;
+  TH2F* denom = get2DHistogram(ibooker,igetter, denomName);
+
+  if (num == NULL)
+    edm::LogWarning("EmDQMPostProcessor") << "2D numerator histogram " << numName << " does not exist";
+
+  if (denom == NULL)
+    edm::LogWarning("EmDQMPostProcessor") << "2D denominator histogram " << denomName << " does not exist";
+
+  // Check if histograms actually exist
+
+  if(!num || !denom) return 0;
+
+  MonitorElement* meOut = ibooker.book2D(outName,titel,num->GetXaxis()->GetNbins(),num->GetXaxis()->GetXmin(),num->GetXaxis()->GetXmax(),num->GetYaxis()->GetNbins(),num->GetYaxis()->GetXmin(),num->GetYaxis()->GetXmax());
```

Comment:
here and elsewhere there are indentation problems. Please check that code in loops is indented relative to the code outside the loop (and code in the same loop/function has the same indentation)